react.md

```diff
@@ -10,37 +10,286 @@ mainmenu: true
-- TODO: How to setup .nvmrc
+## Setting up your workstation
+
+### Node + NVM (Node Version Manager)
+You should install node with [nvm](https://github.com/nvm-sh/nvm), not with Homebrew or the nodejs.org installer. If you've already installed node, please uninstall it before installing nvm. nvm makes it easy to run multiple versions of node side by side on the same machine, allowing you to switch from project to project without running into version issues.
+
+#### .nvmrc
+Each React project should include an [`.nvmrc`](https://github.com/nvm-sh/nvm#nvmrc) with the version of node the project expects.
 
 ### Package management
-Use `npm`, this is a solved problem
+All third-party dependencies should be installed with `npm`. Avoid copying & pasting third-party code directly into the project and avoid using other front-ends to the npm registry like yarn.
+
+All projects should include a `package-lock.json` file to ensure builds are reliable from machine to machine. Please check this file into version control, and avoid deleting & regenerating it. If you run into an issue that you can only solve by regenerating `package-lock.json`, please consult a senior JS developer before doing so to make sure it's absolutely necessary.
+
+## Project Setup
+Use [`create-react-app`](https://facebook.github.io/create-react-app/), React's official setup tool, to bootstrap your new application. CRA provides a bare-bones project structure and automatically configures Webpack, Babel, and the various Webpack loaders necessary to import CSS, SVG, and other assets into your components.
+
+Get started by running the following command, making sure that the name you give your React app matches the name of its git repository.
+```
+npm init react-app $GIT_REPO_NAME --use-npm
+```
 
-## React: Project Setup
-Use `create-react-app`
+### Ejecting
+Depending on project requirements (i.e. supporting more than development, test & production environment configs) you might need to [eject](https://facebook.github.io/create-react-app/docs/available-scripts#npm-run-eject) from CRA and manually configure your project. Consult a senior JS developer if you think you need to do this so we can make sure that ejecting is necessary and your ejected configuration is well documented.
 
-## React: Project Structure
-- TODO: How to set up folder structure
+## Project structure
+We use CRA as a starting point for our structure, but we get a bit more specific about component organization. All of your app code will live inside `src/` as CRA will only compile code in that directory with webpack.
+
+```
+my-app/
+  README.md
+  node_modules/
+  package.json
+  .env
+  public/
+    index.html
+    favicon.ico
+  src/
+    assets/
+      icon.svg
+    containers/
+      App/
+        __tests__/
+          index.test.js
+          SubComponent.test.js
+        index.js
+        SubComponent.js
+        style.module.css
+    components/
+      form/
+      form-partials/
```

Comment:
yes. yes to both folders `form` and `form-partials` - this will help readability so much. I wish we had this two folder structure on Surrey, it would've sped up development quite a bit.